bin/summarize-user-events

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+Dir.chdir(__dir__) { require 'bundler/setup' }
+
+require 'active_support'
+require 'active_support/core_ext/integer/time'
+require 'active_support/core_ext/object/blank'
+require 'active_support/time'
+require 'aws-sdk-cloudwatchlogs'
+require 'concurrent-ruby'
+require 'optparse'
+
+$LOAD_PATH.unshift(File.expand_path(File.join(__dir__, '../lib')))
+require 'reporting/cloudwatch_client'
+require 'reporting/cloudwatch_query_quoting'
+
+# Require all *_matcher.rb files in lib/event_summarizer
+Dir[File.expand_path(
+  File.join(__dir__, '../lib/event_summarizer', '**', '*_matcher.rb'),
+)].sort.each do |f|
+  require f
+end
+
+class SummarizeUserEvents
+  attr_reader :file_name, :uuid, :from_date, :to_date, :zone
+
+  NICE_DATE_AND_TIME_FORMAT = '%B %d, %Y at %I:%M %p %Z'
+  TIME_ONLY_FORMAT = '%I:%M %p'
+
+  def initialize(file_name: nil, user_uuid: nil, start_time: nil, end_time: nil, zone: 'UTC')
+    @file_name = file_name
+    @zone = zone
+    Time.zone = zone
+    @uuid = user_uuid
+    @from_date = parse_time(start_time) || 1.week.ago
+    @to_date = parse_time(end_time) || (start_time.present? ? from_date + 1.week : Time.zone.now)
+  end
+
+  def parse_time(time_str)
+    Time.zone.parse(time_str)
+  rescue StandardError
+    nil
+  end
+
+  def matchers
+    @matchers ||= [
+      EventSummarizer::ExampleMatcher.new,
+      EventSummarizer::AccountDeletionMatcher.new,
+      EventSummarizer::IdvMatcher.new,
+    ]
+  end
+
+  def run
+    find_cloudwatch_events do |event|
+      Time.zone ||= zone
+
+      # Timestamps coming from cloudwatch are in UTC but don't include any zone
+      # information. Here we parse them as UTC.
+      if event['@timestamp'].is_a?(String)
+        event['@timestamp'] = Time.zone.parse("#{event['@timestamp']} UTC")
+      end
+      event['@message'] = JSON.parse(event['@message']) if event['@message'].is_a?(String)
+
+      matchers.each do |matcher|
+        matcher.handle_cloudwatch_event(event)
+      end
+    end
+
+    overall_results = []
+
+    matchers.each do |matcher|
+      results_for_matcher = matcher.finish
+      overall_results.append(*results_for_matcher)
+    end
+
+    puts format_results(overall_results)
```

Comment:
it might make testing easier if you pass in `stdout` to the instance of this class and then provide a `StringIO.new` instance in tests for it?
```suggestion
    stdout.puts format_results(overall_results)
```